be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -815,8 +816,10 @@ void SegmentIterator::_init_current_block(
 void SegmentIterator::_output_non_pred_columns(vectorized::Block* block) {
     SCOPED_RAW_TIMER(&_opts.stats->output_col_ns);
     for (auto cid : _non_predicate_columns) {
-        block->replace_by_position(_schema_block_id_map[cid],
-                                   std::move(_current_return_columns[cid]));
+        auto loc = _schema_block_id_map[cid];
```

Comment:
std::vector<int> _schema_block_id_map; It is a map from schema column id to column idx in Block, so all loc should less than block->columns(), why do we need add this check?
If we need add this check, any else clause needed?

Reply:
have add comment